arroyo/processing/strategies/dead_letter_queue/policies/produce.py

```diff
@@ -0,0 +1,41 @@
+import json
+
+from arroyo.backends.kafka.consumer import KafkaPayload, KafkaProducer
+from arroyo.processing.strategies.dead_letter_queue.policies.abstract import (
+    DeadLetterQueuePolicy,
+    InvalidMessage,
+)
+from arroyo.types import Topic
+from arroyo.utils.metrics import get_metrics
+
+
+class ProduceInvalidMessagePolicy(DeadLetterQueuePolicy):
+    """
+    Produces given InvalidMessage to a dead letter topic.
+
+    Meant to be used as a baseclass for policies needing to produce
+    invalid messages to a dead letter topic.
+    """
+
+    def __init__(self, producer: KafkaProducer, dead_letter_topic: Topic) -> None:
+        self.__metrics = get_metrics()
+        self.__dead_letter_topic = dead_letter_topic
+        self.__producer = producer
+
+    def handle_invalid_message(self, e: InvalidMessage) -> None:
+        """
+        Produces a message to the given dead letter topic in the form:
+
+        {
+            "topic": <original topic the bad message was produced to>,
+            "timestamp": <time at which exception was thrown>,
+            "message": <original bad KafkaMessage>
+        }
+
+        """
+        data = json.dumps(
+            {"topic": e.topic, "timestamp": e.timestamp, "message": e.message}
+        ).encode("utf-8")
+        payload = KafkaPayload(key=None, value=data, headers=[])
```

Comment:
Do we want the the dead letter topic partitioned in any specific way?

Reply:
One partition seems fine for the volume we'd expect to have here. Who should decide what they key or headers should be though? What happens if there is an important key or header that we want to preserve? This line is not keeping those original values.

Reply:
I'm not sure we need abstract out the headers and key. The message thrown could include that info and it could be present in the payload of this new message, but I don't think the headers or key of the message produced to the dead letter topic should reflect the original message.